### ISSUE TITLE
DNS query failed in k8s by service name

### ISSUE THREAD

**Issue by dawnbreaks** [line/armeria#1694]
Hi,  the DnsAddressEndpointGroup can not resolve successfully by the k8s service name, and must use a fully qualified domain name.
e.g. 

```java
//query failed
new DnsAddressEndpointGroupBuilder("service-a").port(8080).build()
//only a fully qualified domain name work
new DnsAddressEndpointGroupBuilder("service-a.infra.svc.cluster.local").port(8080).build()
````

**Comment by trustin** [comment]
Hi! Could you paste your `/etc/hosts` and `/etc/resolv.conf`? IIUC, `/etc/resolv.conf` must contain the following lines:

```
search infra.svc.cluster.local svc.cluster.local cluster.local 
options ndots:5
```

**Comment by dawnbreaks** [comment]
Yes,  the /etc/resolv.conf do contains the search options. It seem that armeria always transform a relative dns name to absulute one by adding a trailing dot at the end of the host name.
```
nameserver 192.168.80.10
search infra.svc.cluster.local svc.cluster.local cluster.local localdomain
options ndots:5
```

**Comment by dawnbreaks** [comment]
Hi, the DefaultDnsQuestion in netty will transform a relative dns name to an absolute one. So the search options will never work.

**Comment by trustin** [comment]
Let us investigate this further. Thanks for reporting. :+1:

**Comment by trustin** [comment]
One more question: Could you set the log level of `io.netty.resolver.dns.TraceDnsQueryLifeCycleObserverFactory` logger to `DEBUG` to see what DNS queries Netty sends?

**Comment by trustin** [comment]
.. and could you also let us know the version of Armeria and Netty in your class path?

**Comment by dawnbreaks** [comment]
armeria version: 0.82.0
netty version:  4.1.33.final

the parent class of DefaultDnsQuestion append the trailing dot to the host name, Please refer to this file [AbstractDnsRecord](https://github.com/netty/netty/blob/4c64c98f348131e0792ba4a92ce3d0003237d56a/codec-dns/src/main/java/io/netty/handler/codec/dns/AbstractDnsRecord.java)

**Comment by trustin** [comment]
Should be fixed with #1697. Please give it a try if you don't mind building Armeria by yourself.

**Comment by dawnbreaks** [comment]
Great. I'm on vacation, and will have a try next week. It 'would be appreciated that An new release that fixed this issue will be made soon.
